lib/davinci_crd_test_kit/client_tests/hook_request_required_fields_test.rb

```diff
@@ -20,21 +20,35 @@ class HookRequestRequiredFieldsTest < Inferno::Test
         a request for
     )
 
-    uses_request :hook_request
-
-    def hook_url
-      base_url + config.options[:hook_path]
-    end
-
     def hook_name
       config.options[:hook_name]
     end
 
-    run do
-      assert_valid_json(request.request_body)
-      request_body = JSON.parse(request.request_body)
+    output :contexts_prefetches
 
-      hook_request_required_fields_check(request_body, hook_name)
+    run do
+      load_tagged_requests(hook_name)
+      skip_if requests.empty?, "No #{hook_name} requests were made in a previous test as expected."
+      error_messages = []
+      contexts_prefetches = []
+      requests.each_with_index do |request, index|
+        assert_valid_json(request.request_body)
+        request_body = JSON.parse(request.request_body)
+        context = request_body['context'] if request_body['context'].is_a?(Hash)
+        prefetch = request_body['prefetch'] if request_body['prefetch'].present? &&
+                                               request_body['prefetch'].is_a?(Hash)
```

Comment:
no need to check the presence here, just checking the class of the object here will return false for nil objects.